woocommerce/woocommerce-ios
# Login: Handle Jetpack site connection for self-hosted sites

Closes #7846 


### Description

When working on #7597 and #7745, we added Jetpack connection handling for accounts without connection to Jetpack sites. 

There's another case where we need to handle Jetpack connection - when a site has not registered with Jetpack. This is when we need to handle site connection before proceeding to account connection.

To successfully register Jetpack connection for a site, we need to request the connection URL, then access it with the same cookies used to request the URL. Otherwise, the URL will be considered expired.

The solution introduced in this PR includes:
- Only show Jetpack error if a user tries to log in to a site that does not have Jetpack installed or activated. If the site only misses Jetpack connection, show them the account mismatch screen, where we can help them register the Jetpack connection on their site.
- Retrieve cookies from WordPressOrgNetwork
- Send retrieved cookies to AuthenticatedWebViewController
- Add a link to show terms and conditions to be agreed on when connecting Jetpack.
- Also: update the primary button title on the Account mismatch screen to "Connect Jetpack to your account" to align with Android.

### Testing instructions

Handling Jetpack connection when logging in with site credentials:
- Pre-requisite: create a JN site with Jetpack installed, activated but not connected. The site may or may not have Woo.
- Log out of the app if needed. On the prologue screen, select Enter site address.
- Enter the address of the JN site.
- Select Log in with site credentials.
- Enter the credentials and tap Continue. Notice that the account mismatch screen is then displayed.
- Notice a new text at the bottom of the screen regarding the terms to be agreed upon when connecting Jetpack.
- Tap Connect Jetpack to your account. Notice that a web view is displayed navigating to the Jetpack authorization screen. Tap Authorize, notice that the app will then navigate to the WP.com login flow and show the home screen after authentication succeeds.

Handling Jetpack connection when logging in with a WP.com account:
- Pre-requisite: create a JN site with Jetpack installed, activated but not connected. The site may or may not have Woo.
- Log out of the app if needed. On the prologue screen, select Enter site address.
- Enter the address of the JN site.
- Enter WP.com credentials and tap Continue. Notice that the account mismatch screen is then displayed.
- Notice a new text at the bottom of the screen regarding the terms to be agreed upon when connecting Jetpack.
- Tap Connect Jetpack to your account. Notice that the site credential login flow is presented. Enter the credentials and tap Continue.
- Notice that a web view is displayed navigating to the Jetpack authorization screen. Tap Authorize, notice that the app will then navigate to the WP.com login flow and show the home screen after authentication succeeds.

Please feel free to test with a mismatch WP.com site. Notice that there is no terms text at the bottom of the account mismatch screen.

### Screenshots


https://user-images.githubusercontent.com/5533851/195795881-12fe325a-fd53-4d37-a910-45837241d9e8.mp4




---
